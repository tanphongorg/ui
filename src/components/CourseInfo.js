import React from 'react'
import { Button, Card, Image } from 'semantic-ui-react'

const CardExample = ({course}) => (
  <Card>
    <Card.Content>
      <Image floated='right' size='mini' src='https://react.semantic-ui.com/assets/images/avatar/large/steve.jpg' />
      <Card.Header>
        { course.title }
      </Card.Header>
      <Card.Meta>
        { course.meta }
      </Card.Meta>
      <Card.Description>
        { course.description }
      </Card.Description>
    </Card.Content>
    <Card.Content extra>
      <div className='ui two buttons'>
        <Button basic color='green'>Approve</Button>
        <Button basic color='red'>Decline</Button>
      </div>
    </Card.Content>
  </Card>
)

export default CardExample
