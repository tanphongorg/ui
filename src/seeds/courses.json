[
  {
    "id": 1,
    "title": "Course 1",
    "meta": "Meta 1",
    "description": "Description 1"
  },
  {
    "id": 2,
    "title": "Course 1",
    "meta": "Meta 1",
    "description": "Description 1"
  },
  {
    "id": 3,
    "title": "Course 1",
    "meta": "Meta 1",
    "description": "Description 1"
  },
  {
    "id": 4,
    "title": "Course 1",
    "meta": "Meta 1",
    "description": "Description 1"
  },
  {
    "id": 5,
    "title": "Course 1",
    "meta": "Meta 1",
    "description": "Description 1"
  },
  {
    "id": 6,
    "title": "Course 1",
    "meta": "Meta 1",
    "description": "Description 1"
  },
  {
    "id": 7,
    "title": "Course 1",
    "meta": "Meta 1",
    "description": "Description 1"
  },
  {
    "id": 8,
    "title": "Course 1",
    "meta": "Meta 1",
    "description": "Description 1"
  },
  {
    "id": 9,
    "title": "Course 1",
    "meta": "Meta 1",
    "description": "Description 1"
  },
  {
    "id": 10,
    "title": "Course 1",
    "meta": "Meta 1",
    "description": "Description 1"
  }
]