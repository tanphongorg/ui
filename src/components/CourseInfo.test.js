/* global it , expect */

import React from 'react'
import CourseInfo from './CourseInfo'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import courses from '../seeds/courses.json'

it('renders without crashing', () => {
  const component = shallow(<CourseInfo course={courses[0]} />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})
