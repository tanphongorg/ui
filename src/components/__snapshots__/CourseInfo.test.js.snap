// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Card>
  <CardContent>
    <Image
      as="img"
      floated="right"
      size="mini"
      src="https://react.semantic-ui.com/assets/images/avatar/large/steve.jpg"
      ui={true}
    />
    <CardHeader>
      Course 1
    </CardHeader>
    <CardMeta>
      Meta 1
    </CardMeta>
    <CardDescription>
      Description 1
    </CardDescription>
  </CardContent>
  <CardContent
    extra={true}
  >
    <div
      className="ui two buttons"
    >
      <Button
        as="button"
        basic={true}
        color="green"
      >
        Approve
      </Button>
      <Button
        as="button"
        basic={true}
        color="red"
      >
        Decline
      </Button>
    </div>
  </CardContent>
</Card>
`;
